jetty-webapp/src/main/java/org/eclipse/jetty/webapp/WebInfConfiguration.java

```diff
@@ -478,18 +478,15 @@ public void resolveTempDirectory(WebAppContext context)
     private File asFile(Object fileattr)
     {
         if (fileattr == null)
-        {
             return null;
-        }
         if (fileattr instanceof File)
-        {
             return (File)fileattr;
-        }
         if (fileattr instanceof String)
-        {
             return new File((String)fileattr);
-        }
-        return null;
+        if (fileattr instanceof Path)
```

Comment:
Need to update the javadoc for this method to say it supports Path type.

I'm not convinced about throwing ISE - the behaviour of this method has always been to return null as the fall through, and the resolveTempDirectory method would go on to make a custom tmp directory. If you throw ISE this will stop the deployment. Maybe just LOG.WARN instead?

Reply:
I think this really should be ISE. There is a big difference between not specifying a temp directory, and specifying one with a random object type that we don't recognise.

Like if someone were to do 
```java
webAppContext.setAttribute(ServletContext.TEMPDIR, new MappedByteBufferPool());
```

should we really ignore this? to me it just seems like a bad configuration that should stop the deployment. 

Just like if we get a File object but the directory isn't writeable or has some permission issue, we will throw ISE and not just ignore and warn. So why should we do something different for this case.

Reply:
Well, the title of this PR is to add extra tests for the tmp directory, not to change behaviour of the tmp directory.  Also, I would add that you should definitely not change the behaviour in jetty-9.4. So only add tests in 9.4, but when you make a separate PR for the merge to 10, you could introduce your behaviour change then (with a better PR title).

Reply:
The title of the PR can easily be edited so that is not an issue. The only case this would break someone is if they have tried to configure a temp directory to be used but it was the wrong type and is being silently ignored.

But anyway I can remove this change from this PR. Should I remove or disable the tests for this? or should I change them to verify that invalid object types for the temp directory are ignored?

Reply:
Your call about the tests for 9.4.  Change the behaviour and the tests for 10/11.

Reply:
I have reverted the behaviour of this and removed the tests.